ui/lib/kv/addon/routes/configuration.js

```diff
@@ -13,8 +13,10 @@ export default class KvConfigurationRoute extends Route {
   model() {
     const backend = this.modelFor('application');
     return hash({
-      mountConfig: this.modelFor('application'),
-      engineConfig: this.store.findRecord('kv/config', backend.id),
+      mountConfig: backend,
+      engineConfig: this.store.findRecord('kv/config', backend.id).catch(() => {
+        return this.store.createRecord('kv/config', { backend: backend.id });
```

Comment:
nit: would be nice to have a note that we're returning a created record so that we still have access to permissions